tests/aws-cpp-sdk-core-tests/aws/auth/CredentialTrackingTest.cpp

```diff
@@ -127,3 +144,325 @@ TEST_F(CredentialTrackingTest, TestEnvironmentCredentialsTracking)
+
+    // Write test credentials to file
+    std::ofstream credFile(credentialsFile.c_str());
+    credFile << "[default]\n";
+    credFile << "aws_access_key_id = test-profile-access-key\n";
+    credFile << "aws_secret_access_key = test-profile-secret-key\n";
+    credFile.close();
+
+    // Set environment to use our test credentials file
+    Aws::Environment::EnvironmentRAII testEnvironment{{
+        {"AWS_SHARED_CREDENTIALS_FILE", credentialsFile},
+    }};
+
+    // Setup mock response
+    std::shared_ptr<HttpRequest> requestTmp =
+        CreateHttpRequest(Aws::Http::URI("dummy"), Aws::Http::HttpMethod::HTTP_POST,
+                        Aws::Utils::Stream::DefaultResponseStreamFactoryMethod);
+    auto successResponse = Aws::MakeShared<Standard::StandardHttpResponse>(ALLOCATION_TAG, requestTmp);
+    successResponse->SetResponseCode(HttpResponseCode::OK);
+    successResponse->GetResponseBody() << "{}";
+    mockHttpClient->AddResponseToReturn(successResponse);
+
+    // Create client configuration
+    Aws::Client::ClientConfigurationInitValues cfgInit;
+    cfgInit.shouldDisableIMDS = true;
+    Aws::Client::ClientConfiguration clientConfig(cfgInit);
+    clientConfig.region = Aws::Region::US_EAST_1;
+
+    // Create credential testing client that uses default provider chain
+    CredentialTestingClient client(clientConfig);
+
+    // Create mock request
+    AmazonWebServiceRequestMock mockRequest;
+
+    // Make request
+    auto outcome = client.MakeRequest(mockRequest);
+    ASSERT_TRUE(outcome.IsSuccess());
+
+    // Verify User-Agent contains profile credentials tracking
+    auto lastRequest = mockHttpClient->GetMostRecentHttpRequest();
+    EXPECT_TRUE(lastRequest.HasHeader(Aws::Http::USER_AGENT_HEADER));
+    const auto& userAgent = lastRequest.GetHeaderValue(Aws::Http::USER_AGENT_HEADER);
+    EXPECT_FALSE(userAgent.empty());
+
+    const auto userAgentParsed = Aws::Utils::StringUtils::Split(userAgent, ' ');
+
+    // Verify there's only one m/ section (no duplicate m/ sections)
+    int mSectionCount = 0;
+    for (const auto& part : userAgentParsed) {
+        if (part.find("m/") != Aws::String::npos) {
+            mSectionCount++;
+        }
+    }
+    EXPECT_EQ(1, mSectionCount);
+
+    // Check for profile credentials business metric (n) in user agent
+    auto businessMetrics = std::find_if(userAgentParsed.begin(), userAgentParsed.end(),
+        [](const Aws::String& value) { return value.find("m/") != Aws::String::npos && value.find("n") != Aws::String::npos; });
+
+    EXPECT_TRUE(businessMetrics != userAgentParsed.end());
+
+    // Cleanup
+    std::remove(credentialsFile.c_str());
+}
+
+TEST_F(CredentialTrackingTest, TestProcessCredentialsTracking)
+{
+    // Create a temporary config file with credential_process
+    Aws::String tempDir = "/tmp";
+    Aws::String awsDir = tempDir + "/.aws";
+    Aws::String configFile = awsDir + "/config";
+
+    // Ensure directory exists
+    mkdir(tempDir.c_str(), 0755);
```

Comment:
dont do it this way, [use temp file and env vars like we do it for the other tests that require a configuration/credentials file](https://github.com/aws/aws-sdk-cpp/blob/main/tests/aws-cpp-sdk-dynamodb-unit-tests/DynamoDBUnitTests.cpp#L425-L433)